build.gradle.kts

```diff
@@ -276,9 +290,40 @@ tasks.shadowJar {
 			"TweakClass" to "org.spongepowered.asm.launch.MixinTweaker",
 		)
 	}
+	
+	doLast {
+		// re-zip the jar with highest compression level
+		val jarFile = shadowJar.archiveFile.get().asFile
+		val fileContents = mutableMapOf<String, ByteArray>()
+		ZipInputStream(jarFile.inputStream()).use { zis ->
+			var entry: ZipEntry? = zis.nextEntry
+			while (entry != null) {
+				if (!entry.isDirectory) {
+					val bytes = zis.readAllBytes()
+					fileContents[entry.name] = bytes
+				}
+				entry = zis.nextEntry
+			}
+		}
+
+		ZipOutputStream(Files.newOutputStream(jarFile.toPath())).use { zos ->
+			zos.setLevel(Deflater.BEST_COMPRESSION)
+			fileContents.forEach { (name, bytes) ->
+				var newBytes = bytes
+				if(name.endsWith(".json") || name.endsWith(".mcmeta") || name.endsWith("mcmod.info")) {
```

Comment:
This is a nitpick but please replace `if(` with `if (`.

Also, should probably just check equality for `mcmod.info` instead of `endsWith`.

Reply:
`.endsWith` is necessary because `name` contains the whole path from the root of the jar. I'll change it to `File.separator + "mcmod.info"` just to be sure though.

Reply:
Oh, in that case the change is not necessary. Whichever way you prefer is fine here.